### PR TITLE
General improvements and fixes

### PR DESCRIPTION
**Pull request by dom96**

### CODE REVIEW

File: readme.rst [hunk 1]
```diff
@@ -7,9 +7,11 @@ to change.
 
 To try it out, run::
 
-  git clone https://github.com/pragmagic/karax
+  cd ~/projects # Insert your favourite directory for projects
+
+  nimble develop # This will clone Karax and create a link to it in ~/.nimble
```
Comment (bluenote10) [inline readme.rst]: Doesn't this require the `git clone` and cd'ing into the cloned directory?

Comment (bluenote10) [reply readme.rst]: I was just confused because the sequence `cd ~/projects && nimble develop` does not mention Karax at all, how should Nimble know that it is supposed to download Karax?

Comment (dom96) [reply readme.rst]: oh, my bad. It should be ``nimble develop karax``, good catch.

File: karax/kajax.nim [hunk 2]
```diff
@@ -38,7 +38,7 @@ proc ajax*(meth, url: cstring; headers: openarray[(cstring, cstring)];
       if this.status == 200:
         contWrapper(this.status, this.responseText)
       else:
-        contWrapper(this.status, this.statusText)
```
Comment (Araq) [inline karax/kajax.nim]: That was done on purpose. Why is 'responseText' better?

File: karax/vdom.nim [hunk 3]
```diff
@@ -267,6 +267,13 @@ when defined(js):
   proc text*(s: string): VNode = VNode(kind: VNodeKind.text, text: kstring(s), index: -1)
 proc text*(s: kstring): VNode = VNode(kind: VNodeKind.text, text: s, index: -1)
 
+when defined(js):
+  proc verbatim*(s: string): VNode =
+    VNode(kind: VNodeKind.verbatim, text: kstring(s), index: -1)
+proc verbatim*(s: kstring): VNode =
```
Comment (Araq) [inline karax/vdom.nim]: Suspicious, why is that required?

File: karax/vdom.nim [hunk 4]
```diff
@@ -267,6 +267,13 @@ when defined(js):
   proc text*(s: string): VNode = VNode(kind: VNodeKind.text, text: kstring(s), index: -1)
 proc text*(s: kstring): VNode = VNode(kind: VNodeKind.text, text: s, index: -1)
 
+when defined(js):
+  proc verbatim*(s: string): VNode =
```
Comment (Araq) [inline karax/vdom.nim]: Suspicious, why is this required?

File: karax/karax.nim [hunk 5]
```diff
@@ -4,6 +4,8 @@ import kdom, vdom, jstrutils, compact, jdict, vstyles
 
 export kdom.Event, kdom.Blob
 
+{.experimental: "notnil".}
```
Comment (Araq) [inline karax/karax.nim]: Wrap that in an ``when defined(nimNoNil)``

File: karax/vdom.nim (hunk 4 above)
Comment (Araq) [reply karax/vdom.nim]: Convenience which ``verbatim`` does not need.
